scalafix-core/shared/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -118,261 +80,369 @@ class EscapeHatch(
     }
 
     val patches =
-      patchesByName.map {
-        case (name, patch) =>
-          loop(name, patch)
-      }.asPatch
-
-    val unusedDisableWarning =
-      LintCategory
-        .warning(
-          "Disable",
-          "This comment does not disable any rule"
-        )
-        .withOwner(unusedScalafixSupression)
-
-    val unusedEscapesWarning =
-      (disableRules -- usedEscapes).values
-        .map(unused => unusedDisableWarning.at(unused.anchorPosition.value))
-
-    val filteredLints = lintMessages.result()
-
-    (patches, filteredLints ++ unusedEnable ++ unusedEscapesWarning)
+      patchesByName.map { case (name, patch) => loop(name, patch) }.asPatch
+
+    // we don't want to show unused warnings for non-Scalafix rules
+    val prefixedEscapes =
+      annotatedEscapes.disableEscapes
+        .filter { escape =>
+          val ruleName = escape.cause.text
+          ruleName.startsWith(AnnotatedEscapes.OptionalRulePrefix)
+        }
+    val unusedDisable =
+      (prefixedEscapes ++ anchoredEscapes.disableEscapes)
+        .filterNot(usedEscapes)
+        .map(_.cause)
+
+    val unusedWarnings =
+      (unusedDisable ++ anchoredEscapes.unusedEnable)
+        .map(UnusedWarning.at)
+    val warnings = lintMessages.result() ++ unusedWarnings
+    (patches, warnings)
   }
 }
 
 object EscapeHatch {
-  private[EscapeHatch] case class EscapeFilter(
+
+  private val UnusedScalafixSuppression = RuleName("UnusedScalafixSuppression")
+  private val UnusedWarning = LintCategory
+    .warning("", "This comment has no effect and can be removed")
+    .withOwner(UnusedScalafixSuppression)
```

Comment:
Actually, the term "comment" is not 100% accurate here given that this warning message can also refer to a rule name in the annotation. Maybe we can say  "escape" or "suppression" instead?

Reply:
Please see if the updated warning message is generic enough for all positions/scenarios where it can refer to:

- specific rule in an annotation:
```
scala/test/escapeHatch/AnnotationPrefixedRules.scala:23:47: error: [UnusedScalafixSuppression]:
Unused Scalafix suppression. This can be removed
  @SuppressWarnings(Array("scalafix:NoNull", "scalafix:NoDummy"))
                                              ^
```

- wildcard ON comment:
```
scala/test/escapeHatch/AnchorMultipleRules.scala:31:3: error: [UnusedScalafixSuppression]:
Unused Scalafix suppression. This can be removed
  // scalafix:on
  ^
```

- wildcard OFF comment:
```
scala/test/escapeHatch/AnchorWildcard.scala:16:3: error: [UnusedScalafixSuppression]:
Unused Scalafix suppression. This can be removed
  // scalafix:off
  ^
```

- specific rule in an ON comment:
```
scala/test/escapeHatch/AnchorMultipleRules.scala:30:18: error: [UnusedScalafixSuppression]:
Unused Scalafix suppression. This can be removed
  /* scalafix:on NoInfer.any2stringadd, Disable.get */
                 ^
```

- specific rule in an OFF comment:
```
scala/test/escapeHatch/AnchorMultipleRules.scala:22:42: error: [UnusedScalafixSuppression]:
Unused Scalafix suppression. This can be removed
 // scalafix:off NoInfer.any2stringadd, Disable.get
                                        ^
```

Reply:
Good catch! "comment" is indeed misleading, I like the new phrasing better 👍